extensions/metrics-cluster-feature/renderer.tsx

```diff
@@ -12,42 +24,73 @@ export default class ClusterMetricsFeatureExtension extends LensRendererExtensio
     if (!cluster.status.active) {
       return;
     }
 
-    const metricsFeature = new MetricsFeature();
+    if (!this.features.has(cluster.getId())) {
+      this.features.set(cluster.getId(), new MetricsFeature(cluster));
+    }
+
+    const metricsFeature = this.features.get(cluster.getId());
 
-    await metricsFeature.updateStatus(cluster);
+    await metricsFeature.updateStatus();
 
     if (metricsFeature.status.installed) {
       if (metricsFeature.status.canUpgrade) {
         ctx.menuItems.unshift({
           icon: "refresh",
           title: "Upgrade Lens Metrics stack",
           onClick: async () => {
-            metricsFeature.upgrade(cluster);
+            try {
+              Component.Notifications.info(`Lens Metrics is being upgraded on ${cluster.metadata.name}`, { timeout: 7_500 });
+
+              await metricsFeature.upgrade();
+              await poll(() => metricsFeature.updateStatus(), ({ installed, canUpgrade }) => installed && !canUpgrade, 5_000);
+
+              Component.Notifications.info(`Lens Metrics has been upgraded on ${cluster.metadata.name}`, { timeout: 7_500 });
+            } catch (error) {
+              Component.Notifications.error(`Lens Metrics failed to be upgraded on ${cluster.metadata.name}: ${error}`);
+            }
           }
         });
       }
       ctx.menuItems.unshift({
         icon: "toggle_off",
         title: "Uninstall Lens Metrics stack",
         onClick: async () => {
-          await metricsFeature.uninstall(cluster);
+          try {
+            Component.Notifications.info(`Lens Metrics is being removed from ${cluster.metadata.name}`, { timeout: 7_500 });
 
-          Component.Notifications.info(`Lens Metrics has been removed from ${cluster.metadata.name}`, { timeout: 10_000 });
+            await metricsFeature.uninstall();
+            await poll(() => metricsFeature.updateStatus(), ({ installed }) => !installed, 5_000);
+
+            Component.Notifications.info(`Lens Metrics has been removed from ${cluster.metadata.name}`, { timeout: 7_500 });
+          } catch (error) {
+            Component.Notifications.error(`Lens Metrics failed to be removed from ${cluster.metadata.name}: ${error}`);
+          }
         }
       });
     } else {
       ctx.menuItems.unshift({
         icon: "toggle_on",
         title: "Install Lens Metrics stack",
         onClick: async () => {
-          metricsFeature.install(cluster);
+          try {
+            Component.Notifications.info(`Lens Metrics is being installed to ${cluster.metadata.name}`, { timeout: 7_500 });
 
-          Component.Notifications.info(`Lens Metrics is now installed to ${cluster.metadata.name}`, { timeout: 10_000 });
+            await metricsFeature.install();
+            await poll(() => metricsFeature.updateStatus(), ({ installed }) => installed, 5_000);
+
+            Component.Notifications.info(`Lens Metrics has been installed to ${cluster.metadata.name}`, { timeout: 7_500 });
```

Comment:
Minor UX thing: Is there a way to dismiss the first notification ("Lens Metrics is being installed...") when posting this one? It looks a little odd/cluttered to see both at the same time (when it takes less than 7.5 seconds to install)

Reply:
Yes we can